trieste/bayesian_optimizer.py

```diff
@@ -446,9 +453,27 @@ def optimize(
         )
 
         history: list[Record[StateType]] = []
+        plot_df: Optional[pd.DataFrame] = None
+
+        summary_writer = logging.get_tensorboard_writer()
+        if summary_writer:
+            with summary_writer.as_default(step=0):
+                logging.text(
+                    "metadata",
+                    f"Number of steps: `{num_steps}`\n\n"
+                    f"Number of initial points: "
+                    f"`{dict((k, len(v)) for k,v in datasets.items())}`\n\n"
+                    f"Acquisition rule:\n\n    {acquisition_rule}\n\n"
+                    f"Models:\n\n    {models}",
```

Comment:
another would be batch size, perhaps some info about the observer as well (e.g. dimensionality, not sure what else can be inferred from the observer, there is no function name there)